k-distribution/include/kframework/builtin/domains.md

```diff
@@ -434,6 +434,12 @@ module MAP-KORE-SYMBOLIC [kore,symbolic]
   rule (_MAP:Map [ K  <-  V1 ]) [ K ]  => V1 [simplification]
   rule ( MAP:Map [ K1 <- _V1 ]) [ K2 ] => MAP [ K2 ] requires K1 =/=K K2 [simplification]
 
+  rule (K  |->  V M:Map) [  K ] orDefault _ => V ensures notBool (K in_keys(M)) [simplification]
+  rule (K1 |-> _V M:Map) [ K2 ] orDefault D => M [K2] orDefault D requires K1 =/=K K2 ensures notBool (K1 in_keys(M)) [simplification]
+  rule (_MAP:Map [ K  <-  V1 ]) [ K ] orDefault _ => V1 [simplification]
+  rule ( MAP:Map [ K1 <- _V1 ]) [ K2 ] orDefault D => MAP [ K2 ] orDefault D requires K1 =/=K K2 [simplification]
+  rule .Map [ _ ] orDefault D => D [simplification]
```

Comment:
One small generalization of this rule, which should enable both this simpliifcation and many others (as well as actually exercising some of the new tests), without being considered an "extension" of the previous lemmass because there is no analogue in the previous lemmsa for this:

```suggestion
  rule M [ K ] orDefault D => D requires notBool K in_keys(M) [simplification]
```

This would let it simplify out more, at the expense of triggering more frequent `in_keys(...)` checks.

Reply:
Per discussion with Everett over slack we decided to leave this lemma as-is for now since it's current form suffices for the C semantics example I was working with. We can  revisit Everett's suggestion if we discover terms that we want to simplify that these lemmas are not powerful enough to handle.